### PR TITLE
Fix binance import error - Closes #793

### LINKED ISSUE #793: Binance import error
### Actual behavior

During refreshing the data source from Binance the following error occurs:

```
 2021-09-24T09:23:05.118 ERROR [binance] TypeError: Cannot read property 'data' of undefined
     at fetchAllMarketTickers (/home/lisk/lisk-service/market/shared/market/sources/binance.js:52:19)
     at async reload (/home/lisk/lisk-service/market/shared/market/sources/binance.js:100:19)
     at async Task.controller [as execution] (/home/lisk/lisk-service/market/jobs/binance.js:30:4)
 TypeError: tickers.filter is not a function
     at filterTickers (/home/lisk/lisk-service/market/shared/market/sources/binance.js:62:34)
     at reload (/home/lisk/lisk-service/market/shared/market/sources/binance.js:101:27)
     at async Task.controller [as execution] (/home/lisk/lisk-service/market/jobs/binance.js:30:4)
```

### Expected behavior

Make sure the data from Binance is retrieved without problems.

### Steps to reproduce

Start the process and observe logs.

### Which version(s) does this affect? (Environment, OS, etc...)

`0.5.0-alpha.0`

### PR DESCRIPTION
**Pull request by nagdahimanshu**
### What was the problem?
This PR resolves #793 

### How was it solved?
- [x] Throw error in case of `undefined` response from exchanges (Binance, Bittrex, and Kraken)
- [x] Add array check for market tickers

### How was it tested?
Local + Docker

### CODE REVIEW

File: services/market/shared/market/sources/bittrex.js [hunk 1]
```diff
@@ -37,12 +37,13 @@ const symbolMap = {
 const fetchAllMarketTickers = async () => {
 	try {
 		const response = await requestLib(`${apiEndpoint}/markets/tickers`);
+		if (response === undefined) throw new Error('Data from Bittrex is unavailable');
```
Comment (MichalTuleja) [inline services/market/shared/market/sources/bittrex.js:40]: Does it mean the error message with a stack trace will be displayed when Bittrex is unavailable? I think it deserves only a mention in the custom warning message.

File: services/market/shared/market/sources/binance.js [hunk 2]
```diff
@@ -48,12 +48,13 @@ const symbolMap = {
 const fetchAllMarketTickers = async () => {
 	try {
 		const response = await requestLib(`${apiEndpoint}/ticker/price`);
+		if (response === undefined) throw new Error('Data from Binance is unavailable');
```
Comment (MichalTuleja) [inline services/market/shared/market/sources/binance.js:51]: Same concern as the Bittrex comment.

File: services/market/shared/market/sources/bittrex.js (hunk 1 above)
Comment (sameersubudhi) [reply services/market/shared/market/sources/bittrex.js:40]: @nagdahimanshu Please make the changes within the jobs as we discussed.

Comment (nagdahimanshu) [reply services/market/shared/market/sources/bittrex.js]: Done!

**Review by MichalTuleja — CHANGES_REQUESTED** [review]
There are still errors with stack trace:

```
market_1            | 2022-02-03T12:15:05.107 ERROR [binance] Data from Binance is currently unavailable
market_1            | 2022-02-03T12:15:05.108 ERROR [binance] ServiceUnavailableException: Data from Binance is currently unavailable
market_1            |     at fetchAllMarketTickers (/home/lisk/lisk-service/market/shared/market/sources/binance.js:56:37)
market_1            |     at async reload (/home/lisk/lisk-service/market/shared/market/sources/binance.js:106:19)
market_1            |     at async Task.controller [as execution] (/home/lisk/lisk-service/market/jobs/binance.js:44:4)
market_1            | 2022-02-03T12:15:05.109 WARN [binance] Unable to fetch market prices from Binance right now. Will retry later.
market_1            | ServiceUnavailableException: Data from Binance is currently unavailable
market_1            |     at fetchAllMarketTickers (/home/lisk/lisk-service/market/shared/market/sources/binance.js:56:37)
market_1            |     at async reload (/home/lisk/lisk-service/market/shared/market/sources/binance.js:106:19)
market_1            |     at async Task.controller [as execution] (/home/lisk/lisk-service/market/jobs/binance.js:44:4)
```

Please update it to avoid throwing an error.

File: services/market/jobs/binance.js [hunk 3]
```diff
@@ -13,21 +13,35 @@
...
 
+const logger = Logger();
+
+const reloadMarketPrices = async () => reload()
+	.catch(err => {
+		if (err instanceof ServiceUnavailableException) {
+			logger.warn('Unable to fetch market prices from Binance right now. Will retry later.');
+		}
+		throw err;
+	});
```
Comment (sameersubudhi) [inline services/market/jobs/binance.js:31]: Kindly ensure that the control flow ends after `ServiceUnavailableException` is handled. Same for the other jobs.

```suggestion
const reloadMarketPrices = async () => reload()
	.catch(err => {
		if (err instanceof ServiceUnavailableException) {
			logger.warn('Unable to fetch market prices from Binance right now. Will retry later.');
			return;
		}
		throw err;
	});
```
or
```suggestion
const reloadMarketPrices = async () => reload()
	.catch(err => {
		if (err instanceof ServiceUnavailableException) {
			logger.warn('Unable to fetch market prices from Binance right now. Will retry later.');
		} else {
			throw err;
		}
	});
```

**Review by sameersubudhi — APPROVED** [review]
LGTM 👍

**Review by MichalTuleja — APPROVED** [review]

**Pull request merged**